pkg/utils/authcheck/authchek_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authcheck
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/google/knative-gcp/pkg/logging"
+)
+
+func TestAuthenticationCheck(t *testing.T) {
+	testCases := []struct {
+		name           string
+		authType       AuthType
+		wantStatusCode int
+		setEnv         bool
+	}{
+		{
+			name:           "authentication check uses empty authType",
+			authType:       "",
+			wantStatusCode: http.StatusUnauthorized,
+			setEnv:         false,
+		},
+		{
+			name:           "authentication check uses secret authType",
+			authType:       Secret,
+			wantStatusCode: http.StatusUnauthorized,
+			setEnv:         true,
+		},
```

Comment:
Add a test for WI-gsa as well. Likely by making the http client a var and swapping it out during the test.